### DISCUSSION TITLE
What is considered a failure to trigger a retry?

### DISCUSSION THREAD

**Discussion by creasewp** [linkerd/linkerd2#7854] [Q&A] [answered]
I have been unable to find any documentation that indicates what exactly will be considered a failure that will trigger a retry. I assume things like a timeout or pod being unavailable would be considered a failure, but what about HTTP status codes? Through testing, I figured out that HTTP 500 is a failure, but what about others? Are all 5xx errors considered failures? What about 4xx?

**Comment by mateiidavid** [comment, marked as answer]
Hey @creasewp, by default only 5xx status codes are considered failures; a 403 for example (`Forbidden`) would be classed as a success. With that being said, you can overwrite the defaults on a service-by-service basis using [ServiceProfiles](https://linkerd.io/2.10/reference/service-profiles/#response-class) and adding in a `responseClass` object.
